tutorials/RandomNumberGeneration/Tests.qs

```diff
@@ -17,65 +17,108 @@ namespace Quantum.Kata.RandomNumberGeneration {
         Message("Testing N = 1...");
-        CheckFlatDistribution(RandomNBits, 1, 0.4, 0.6, 1000, 450);
+        RetryCheckUniformDistribution(RandomNBits_Wrapper, 0, 1, 1000, 3);
+        Message("Test Passed Succesfully for N=1 \n");
+
         Message("Testing N = 2...");
-        CheckFlatDistribution(RandomNBits, 2, 1.4, 1.6, 1000, 200);
-        Message("Testing N = 3...");
-        CheckFlatDistribution(RandomNBits, 3, 3.3, 3.7, 1000, 90);
-        Message("Testing N = 10...");
-        CheckFlatDistribution(RandomNBits, 10, 461.0, 563.0, 1000, 0);
+        RetryCheckUniformDistribution(RandomNBits_Wrapper, 0, 3, 1000, 3);
+        Message("Test Passed Succesfully for N=2 \n");
+
+        Message("\n Testing N = 3...");
+        RetryCheckUniformDistribution(RandomNBits_Wrapper, 0, 7, 1000, 3);
+        Message("Test Passed Succesfully for N=3 \n");
+
+        Message("\n Testing N = 10...");
+        RetryCheckUniformDistribution(RandomNBits_Wrapper, 0, 1023, 1000, 3);
+        Message("Test Passed Succesfully for N=10 \n");
+    }
+
+    operation RandomNBits_Wrapper (min: Int, max: Int) : Int {
+        // For N bit random number : min = 0, max = 2^N - 1
+        let N = Ceiling( Lg(IntAsDouble(max+1)) );
+        return RandomNBits(N);
+    }
+
+    operation RetryCheckUniformDistribution( fTest: ((Int,Int)=>Int), min: Int, max: Int, nRuns: Int, numRetries: Int) : Unit {
+        mutable sufficientlyRandom = false;
+        mutable attemptNum = 1;
+        repeat
+        {
+            Message($"attemptNum = {attemptNum}");
+            set sufficientlyRandom = CheckUniformDistribution(fTest, min, max, nRuns);
+            set attemptNum += 1;
+        } until( sufficientlyRandom == true or attemptNum >= numRetries );
+
+        if(sufficientlyRandom == false)
+        {
+            fail $"Failed to generate a random integer as required";
+        }
     }
 
     /// # Input
     /// ## f
     /// Random number generation operation to be tested
     /// ## numBits
     /// Number of bits in the generated result
-    /// ## lowRange
-    /// The lower bound of the median and average for generated dataset
-    /// ## highRange
-    /// The upper bound of the median and average for generated dataset
     /// ## nRuns
     /// The number of random numbers to generate for test
-    /// ## minimumCopiesGenerated
-    /// The minimum number of times each possible number should be generated
-    operation CheckFlatDistribution (f : (Int => Int), numBits : Int, lowRange : Double, highRange : Double, nRuns : Int, minimumCopiesGenerated : Int) : Unit {
-        let max = PowI(2, numBits);
-        mutable counts = ConstantArray(max, 0);
+    operation CheckUniformDistribution (f : ((Int, Int) => Int), min : Int, max : Int, nRuns : Int) : Bool {
+        let idealMean = 0.5 * IntAsDouble(max + min) ;
+        let rangeDividedByTwo = 0.5 * IntAsDouble(max - min);
+        /// variance = a*(a+1)/3, where a= (max=min)/2
```

Comment:
```suggestion
        // variance = a*(a+1)/3, where a = (max-min)/2
```
`///` are documentation comments used to generate quick help for operations and data types. Inside the operation they'll still work because `//` is a regular comment, but the third slash is not really required

Reply:
(The same suggestion in other places where `///` comment is used inside operation code)

Reply:
Done